src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java

```diff
@@ -91,7 +91,7 @@ protected DetailAST getListChild() {
     protected IndentLevel getChildrenExpectedIndent() {
         final IndentLevel expectedIndent =
             new IndentLevel(getIndent(), getIndentCheck().getArrayInitIndent(),
-                    getIndentCheck().getLineWrappingIndentation());
+                getLineWrappingIndentation());
```

Comment:
Having issue fix and big code simplification changes in 1 commit may make this too hard to review.
Unless @romani wants otherwise, I think we should split this into 2 commits.
1st commit is minor with all code simplifications.
2nd commit is issue with all code related to issue fix.